typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -7,243 +7,237 @@ import { sleep, strip0x } from '@hyperlane-xyz/utils';
   }
 
-  async verifyChain(
-    chain: ChainName,
-    inputs: VerificationInput,
-  ): Promise<void> {
-    this.logger(`Verifying ${chain}...`);
-    for (const input of inputs) {
-      await this.verifyContract(chain, input);
-    }
-  }
-
   private async submitForm(
     chain: ChainName,
     action: ExplorerApiActions,
-    options?: Record<string, string>,
+    verificationLogger: Debugger,
+    options?: FormOptions<typeof action>,
   ): Promise<any> {
     const { apiUrl, family } = this.multiProvider.getExplorerApi(chain);
-    const isGetRequest =
-      action === ExplorerApiActions.CHECK_STATUS ||
-      action === ExplorerApiActions.CHECK_PROXY_STATUS ||
-      action === ExplorerApiActions.GETSOURCECODE;
-    const params = new URLSearchParams({
-      ...(this.apiKeys[chain] ? { apikey: this.apiKeys[chain] } : {}),
-      module: 'contract',
-      action,
-      ...options,
-    });
-
-    let response: Response;
-    if (isGetRequest) {
-      response = await fetch(`${apiUrl}?${params}`);
-    } else {
-      // For Blockscout explorers, we need to ensure module and action are always query params
-      if (family === ExplorerFamily.Blockscout) {
-        const formData = new URLSearchParams();
-        const urlWithParams = new URL(apiUrl);
-        urlWithParams.searchParams.append('module', 'contract');
-        urlWithParams.searchParams.append('action', action);
-
-        // remove any extraneous keys from body
-        for (const [key, value] of params) {
-          switch (key) {
-            case 'module':
-            case 'action':
-            case 'licenseType':
-            case 'apikey':
-              break;
-            default:
-              formData.append(key, value);
-              break;
-          }
-        }
+    const params = new URLSearchParams();
+    params.set('module', 'contract');
+    params.set('action', action);
 
-        response = await fetch(urlWithParams.toString(), {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded',
-          },
-          body: formData,
-        });
-      } else {
-        response = await fetch(apiUrl, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded',
-          },
-          body: params,
-        });
+    // no need to provide every argument for every request
+    if (options) {
+      for (const [key, value] of Object.entries(options)) {
+        params.set(key, value);
       }
     }
 
-    let result;
-    let responseText;
-    try {
-      responseText = await response.text();
-      result = JSON.parse(responseText);
-    } catch (e) {
-      this.logger(
-        `[${chain}] Failed to parse response from ${responseText}`,
-        e,
-      );
+    // only include apikey if provided & not blockscout
+    if (family !== ExplorerFamily.Blockscout && this.apiKeys[chain]) {
+      params.set('apikey', this.apiKeys[chain]);
```

Comment:
interesting, does blockscout not have API keys?

Reply:
can't find a way to inevm/manta as they don't seem to have accounts functionalitiy, but others we don't use like optimism do

(note: eth blockscout + gnosis testnet blockscout support let you create accounts/api keys)

there's also some further work we need on the keys side to organise by explorer type, as currently we can only store one key per network and we don't know the explorer family this key belongs to